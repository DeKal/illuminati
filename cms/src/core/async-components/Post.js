import loadable from '@loadable/component'

const Post = loadable(() =>
  import(/* webpackChunkName: "post" */ 'post/containers/Post')
)

export default Post
