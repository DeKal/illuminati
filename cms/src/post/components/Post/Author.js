import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/TextField'

const Author = ({ isEdited, author, editPostAuthor }) => (
  <TextField
    label="Author"
    variant="outlined"
    value={author}
    fullWidth
    error={isEdited}
    onChange={event => editPostAuthor(event.target.value)}
  />
)

export default Author

Author.propTypes = {
  isEdited: PropTypes.bool,
  author: PropTypes.string,
  editPostAuthor: PropTypes.func
}
