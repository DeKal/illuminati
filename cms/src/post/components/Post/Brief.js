import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/TextField'

const Brief = ({ isEdited, brief, editPostBrief }) => (
  <TextField
    label="Post Brief"
    required
    variant="outlined"
    value={brief}
    multiline
    rows={6}
    rowsMax={8}
    fullWidth
    error={isEdited}
    onChange={event => editPostBrief(event.target.value)}
  />
)

export default Brief

Brief.propTypes = {
  isEdited: PropTypes.bool,
  brief: PropTypes.string,
  editPostBrief: PropTypes.func
}
