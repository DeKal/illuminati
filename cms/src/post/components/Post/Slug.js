import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/TextField'

const Slug = ({ isEdited, slug, editPostSlug }) => (
  <TextField
    label="Slug"
    required
    variant="outlined"
    value={slug}
    fullWidth
    error={isEdited}
    onChange={event => editPostSlug(event.target.value)}
  />
)

export default Slug

Slug.propTypes = {
  isEdited: PropTypes.bool,
  slug: PropTypes.string,
  editPostSlug: PropTypes.func
}
