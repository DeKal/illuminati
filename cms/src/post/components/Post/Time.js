import React from 'react'
import PropTypes from 'prop-types'
import { MuiPickersUtilsProvider } from '@material-ui/pickers'
import DateFnsUtils from '@date-io/date-fns'
import DatePicker from 'post/components/DatePicker'

const Time = ({ isEdited, time, editPostTime }) => {
  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <DatePicker
        disableToolbar
        format="dd/MM/yyyy"
        label="Time"
        required
        variant="inline"
        margin="normal"
        fullWidth
        error={isEdited}
        value={time}
        onChange={editPostTime}
        KeyboardButtonProps={{
          'aria-label': 'change date'
        }}
      />
    </MuiPickersUtilsProvider>
  )
}

export default Time

Time.propTypes = {
  isEdited: PropTypes.bool,
  time: PropTypes.oneOfType([PropTypes.node, PropTypes.instanceOf(Date)]),
  editPostTime: PropTypes.func
}
