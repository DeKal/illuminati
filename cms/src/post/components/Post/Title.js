import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/TextField'

const Title = ({ isEdited, title, editPostTitle }) => (
  <TextField
    label="Post title"
    required
    variant="outlined"
    value={title}
    multiline
    rowsMax={2}
    fullWidth
    error={isEdited}
    onChange={event => editPostTitle(event.target.value)}
  />
)

export default Title

Title.propTypes = {
  isEdited: PropTypes.bool,
  title: PropTypes.string,
  editPostTitle: PropTypes.func
}
