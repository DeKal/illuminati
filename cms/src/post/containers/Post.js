import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { useParams } from 'react-router-dom'
import { fetchPost, saveEditedContent } from 'post/state/actions'
import { getIsPostLoading } from 'post/state/selectors'

import Post from 'post/components/Post'

const mapStateToProps = state => ({
  loading: getIsPostLoading(state)
})

const mapDispatchToProps = {
  fetchPost,
  saveEditedContent
}

const PostWithHook = props => {
  const { fetchPost } = props
  const { slug } = useParams()

  useEffect(() => {
    fetchPost(slug)
  }, [slug, fetchPost])
  return <Post {...props} />
}
PostWithHook.propTypes = {
  fetchPost: PropTypes.func
}

export default connect(mapStateToProps, mapDispatchToProps)(PostWithHook)
