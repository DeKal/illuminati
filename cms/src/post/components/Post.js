import React from 'react'
import PropTypes from 'prop-types'
import Typography from '@material-ui/core/Typography'

import Container from 'core/components/Container'
import Paper from 'core/components/Paper'
import Grid from 'post/components/Grid'

import Title from 'post/containers/Title'
import Slug from 'post/containers/Slug'
import Author from 'post/containers/Author'
import Time from 'post/containers/Time'
import Brief from 'post/containers/Brief'
import Tags from 'post/containers/Tags'
import Content from 'post/containers/Content'
import PostButton from 'post/components/Button'
import Loading from 'core/components/Loading'

const Post = ({ loading, saveEditedContent }) => {
  if (loading) {
    return <Loading />
  }
  return (
    <Container>
      <Paper>
        <Grid container style={{ padding: '16px' }} spacing={3}>
          <Grid item sm={11}>
            <Typography variant="h6">Editing Post</Typography>
          </Grid>
        </Grid>
        <Grid container>
          <Grid container item direction="column" xs={12} sm={8} spacing={3}>
            <Grid item>
              <Title />
            </Grid>
            <Grid item>
              <Brief />
            </Grid>
            <Grid item>
              <Content />
            </Grid>
          </Grid>
          <Grid container item direction="column" xs={6} sm={4} spacing={3}>
            <Grid item>
              <Slug />
            </Grid>
            <Grid item>
              <Author />
            </Grid>
            <Grid item>
              <Time />
            </Grid>
            <Grid item>
              <Tags />
            </Grid>
            <Grid item>
              <PostButton
                variant="contained"
                size="large"
                color="primary"
                onClick={saveEditedContent}
              >
                Save
              </PostButton>
            </Grid>
          </Grid>
        </Grid>
      </Paper>
    </Container>
  )
}

Post.propTypes = {
  loading: PropTypes.bool,
  saveEditedContent: PropTypes.func
}

export default Post
