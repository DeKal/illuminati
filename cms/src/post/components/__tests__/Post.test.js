import React from 'react'
import { shallow } from 'enzyme'
import Post from 'post/components/Post'

describe('Post', () => {
  it('should render Loading when loading correctly', function() {
    const wrap = shallow(<Post loading />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render Post correctly with editing post', function() {
    const wrap = shallow(<Post loading={false} />)
    expect(wrap).toMatchSnapshot()
  })

  it('should trigger saveEditedContent when click save button', function() {
    const saveEditedContent = jest.fn()
    const wrap = shallow(
      <Post loading={false} saveEditedContent={saveEditedContent} />
    )
    wrap.find('Styled(WithStyles(ForwardRef(Button)))').simulate('click')
    expect(saveEditedContent.mock.calls.length).toBe(1)
  })

  it('should render Post correctly with create post', function() {
    const wrap = shallow(<Post loading={false} newPost />)
    expect(wrap).toMatchSnapshot()
  })

  it('should trigger createPost when click save button', function() {
    const createPost = jest.fn()
    const wrap = shallow(
      <Post loading={false} createPost={createPost} newPost />
    )
    wrap.find('Styled(WithStyles(ForwardRef(Button)))').simulate('click')
    expect(createPost.mock.calls.length).toBe(1)
  })
})
