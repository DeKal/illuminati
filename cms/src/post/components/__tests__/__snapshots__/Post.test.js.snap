// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render Loading when loading correctly 1`] = `<Loading />`;

exports[`Post should render Post correctly with create post 1`] = `
<Styled(div)>
  <WithStyles(WithStyles(ForwardRef(Paper)))>
    <WithStyles(WithStyles(ForwardRef(Grid)))
      container={true}
      spacing={3}
      style={
        Object {
          "padding": "16px",
        }
      }
    >
      <WithStyles(WithStyles(ForwardRef(Grid)))
        item={true}
        sm={11}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h6"
        >
          New Post
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(WithStyles(ForwardRef(Grid)))>
    </WithStyles(WithStyles(ForwardRef(Grid)))>
    <WithStyles(WithStyles(ForwardRef(Grid)))
      container={true}
    >
      <WithStyles(WithStyles(ForwardRef(Grid)))
        container={true}
        direction="column"
        item={true}
        sm={8}
        spacing={3}
        xs={12}
      >
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Title) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Brief) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <ForwardRef />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
      </WithStyles(WithStyles(ForwardRef(Grid)))>
      <WithStyles(WithStyles(ForwardRef(Grid)))
        container={true}
        direction="column"
        item={true}
        sm={4}
        xs={6}
      >
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Slug) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Author) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Time) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(TagsWithStyle) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Styled(WithStyles(ForwardRef(Button)))
            color="primary"
            onClick={[Function]}
            size="large"
            variant="contained"
          >
            Save
          </Styled(WithStyles(ForwardRef(Button)))>
        </WithStyles(WithStyles(ForwardRef(Grid)))>
      </WithStyles(WithStyles(ForwardRef(Grid)))>
    </WithStyles(WithStyles(ForwardRef(Grid)))>
  </WithStyles(WithStyles(ForwardRef(Paper)))>
</Styled(div)>
`;

exports[`Post should render Post correctly with editing post 1`] = `
<Styled(div)>
  <WithStyles(WithStyles(ForwardRef(Paper)))>
    <WithStyles(WithStyles(ForwardRef(Grid)))
      container={true}
      spacing={3}
      style={
        Object {
          "padding": "16px",
        }
      }
    >
      <WithStyles(WithStyles(ForwardRef(Grid)))
        item={true}
        sm={11}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h6"
        >
          Editing Post
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(WithStyles(ForwardRef(Grid)))>
    </WithStyles(WithStyles(ForwardRef(Grid)))>
    <WithStyles(WithStyles(ForwardRef(Grid)))
      container={true}
    >
      <WithStyles(WithStyles(ForwardRef(Grid)))
        container={true}
        direction="column"
        item={true}
        sm={8}
        spacing={3}
        xs={12}
      >
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Title) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Brief) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <ForwardRef />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
      </WithStyles(WithStyles(ForwardRef(Grid)))>
      <WithStyles(WithStyles(ForwardRef(Grid)))
        container={true}
        direction="column"
        item={true}
        sm={4}
        xs={6}
      >
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Slug) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Author) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(Time) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Connect(TagsWithStyle) />
        </WithStyles(WithStyles(ForwardRef(Grid)))>
        <WithStyles(WithStyles(ForwardRef(Grid)))
          item={true}
        >
          <Styled(WithStyles(ForwardRef(Button)))
            color="primary"
            onClick={[Function]}
            size="large"
            variant="contained"
          >
            Save
          </Styled(WithStyles(ForwardRef(Button)))>
        </WithStyles(WithStyles(ForwardRef(Grid)))>
      </WithStyles(WithStyles(ForwardRef(Grid)))>
    </WithStyles(WithStyles(ForwardRef(Grid)))>
  </WithStyles(WithStyles(ForwardRef(Paper)))>
</Styled(div)>
`;
